src/Store.js

```diff
@@ -2,42 +2,114 @@ import {
   observable,
 } from 'mobx';
 
-class EmptyTransporter {}
+import DefaultItem from './Item';
+import EmptyTransporter from './Transporters/EmptyTransporter';
+import { ROLE, SOURCE, STATE } from './constants';
 
 export default class Store {
 
-  static transporterOptions = {
-    http: {
-      url: 'http',
-    },
+  @observable items = [];
+  @observable loaded = false;
+  // @observable deletedItems = [];
+  constructor({
+    Item = DefaultItem, schema,
+    transporter = new EmptyTransporter(),
+    clientStorage = new EmptyTransporter(),
+    options = { autoSave: true },
+  }) {
+    if (!schema) {
+      throw new Error('undefined schema');
+    }
+    this.transporter = transporter;
+    this.transporter.setEnvironment({ store: this, role: ROLE.TRANSPORTER });
+    this.clientStorage = clientStorage;
+    this.transporter.setEnvironment({ store: this, role: ROLE.CLIENT_STORAGE });
+    this.schema = schema;
+    this._options = options;
+    this._Item = Item;
+
+    this._isLoaded = {};
+    this._isLoaded.promise = new Promise((resolve, reject) => {
+      this._isLoaded.resolve = resolve;
+      this._isLoaded.reject = reject;
+    });
+
+    this.clientStorage.initialFetch([])
+    .then((csData) => {
+      csData.items.forEach((item) => {
+        item._transporterState = this._castItemTransporterState(item._transporterState);
+      });
+      this._createItems(csData.items, SOURCE.CLIENT_STORAGE);
+      return this.transporter.initialFetch(csData.items);
```

Comment:
I find the flow more readable and understandable when this "then" function would just return csData.items
and transporter.initialFetch would be done in a seperate then like this:

```js
.then((csData) => {
   // ...
   return csData.items;
})
.then((items) => this.transporter.initialFetch(items))
.then((tData) => {
   // ...
});
```

Because this way each then block has its own job